functional-tests/test-bed/src/main/java/org/arquillian/smart/testing/ftest/testbed/rules/TestBed.java

```diff
@@ -101,15 +106,15 @@ private void after() throws Exception {
     }
 
     private String targetRepoPerTestFolder(Description description) {
-        return GitClone.GIT_REPO_FOLDER
+        return gitClone.getGitRepoFolder()
```

Comment:
> we can have problems when run in parallel threads.

We do have or we can have?

Why would it be a problem? `GitClone#before` is called as a class rule so it should be invoked only once.